src/utility/include/mp-units/math.h

```diff
@@ -136,6 +136,30 @@ template<auto R, typename Rep>
   return {static_cast<Rep>(abs(q.numerical_value_ref_in(q.unit))), R};
 }
 
+/**
+ * @brief Computes the fma of 3 quantities
+ *
+ * @param a: Multiplicand
+ * @param x: Multiplicand
+ * @param b: Addend
+ * @return Quantity: The nearest floating point representable to ax+b
+ */
+template<auto R, auto S, auto T, typename Rep>
+[[nodiscard]] constexpr quantity<T, Rep> fma(const quantity<R, Rep>& a, const quantity<S, Rep>& x,
+                                             const quantity<T, Rep>& b) noexcept
+  requires requires {
+    fma(a.numerical_value_ref_in(a.unit), x.numerical_value_ref_in(x.unit), b.numerical_value_ref_in(b.unit));
+  } || requires {
+    std::fma(a.numerical_value_ref_in(a.unit), x.numerical_value_ref_in(x.unit), b.numerical_value_ref_in(b.unit));
+  }
+{
+  using std::fma;
+  return {static_cast<Rep>(
+            fma(a.numerical_value_ref_in(a.unit), x.numerical_value_ref_in(x.unit), b.numerical_value_ref_in(b.unit))),
+          T};
```

Comment:
From reading your comment (<https://github.com/mpusz/mp-units/issues/525#issuecomment-1841204125>):
> @y9rabbito : In my case, the arguments to fma are _not_ the same type, to wit, `x` is dimensionless and the coefficients are dimensioned. In general, the only requirement I see is that in `fma(a, x, b)`, [a][x] == [b].

It seems like the return type should be a dimensionless quantity.
And all parameters just `Quantity auto`.
Unlike other functions in `math.h`, it seems like a totally unconstrained function.
In fact, you have an use case where even `Quantity auto` wouldn't work.